### PR TITLE
feat: implement getBooks() to retrieve an owner's books from firestore

### PR DESCRIPTION
**Pull request by nayancprakash**
### Description of changes
* implements the `getBooks()` function within `MyBooksActivity` to retrieve an Owner's books from FireStore

### CODE REVIEW

File: src/app/src/main/java/com/example/bookmark/MyBooksActivity.java [hunk 1]
```diff
@@ -76,20 +83,30 @@ public void onItemClick(AdapterView<?> parent, View view,
...
-        allBooks.add(b1);
-        allBooks.add(b2);
+        String username = UserUtil.getLoggedInUser(this);
+        StorageServiceProvider.getStorageService().retrieveUserByUsername(
+            username,
+            user -> {
+                OnSuccessListener<List<Book>> onBooksSuccessListener = new OnSuccessListener<List<Book>>() {
+                    @Override
+                    public void onSuccess(List<Book> books) {
+                        System.out.println("hello");
```
Comment (KonradStanski) [inline src/app/src/main/java/com/example/bookmark/MyBooksActivity.java:93]: Should we leave this in?

**Review by KonradStanski — DISMISSED** [review]

File: src/app/src/main/java/com/example/bookmark/MyBooksActivity.java (hunk 1 above)
Comment (kylehennig) [reply src/app/src/main/java/com/example/bookmark/MyBooksActivity.java:93]:
```suggestion
```

Comment (nayancprakash) [reply src/app/src/main/java/com/example/bookmark/MyBooksActivity.java:93]: no we shouldn't, nice catch

**Review by KonradStanski — DISMISSED** [review]

File: src/app/src/main/java/com/example/bookmark/MyBooksActivity.java [hunk 2]
```diff
@@ -76,20 +83,30 @@ public void onItemClick(AdapterView<?> parent, View view,
...
+                        booksAdapter.notifyDataSetChanged();
+                    }
+                };
+                OnFailureListener onBooksFailureListener = new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+
+                    }
+                };
+                StorageServiceProvider.getStorageService().retrieveBooksByOwner(user, onBooksSuccessListener, onBooksFailureListener);
```
Comment (kylehennig) [inline src/app/src/main/java/com/example/bookmark/MyBooksActivity.java:105]:
```suggestion
                StorageServiceProvider.getStorageService().retrieveBooksByOwner(user, onBooksSuccessListener, e -> DialogUtil.showErrorDialog(this, e));
```

**Review by kylehennig — APPROVED** [review]

**Review by KonradStanski — APPROVED** [review]

**Pull request merged**